File: Programmers/Lv.1/소수만들기.py
def isPrime(a, b, c):
    total = a + b + c

    for i in range(2, total): 
        if total % i == 0 : return False 
    return True 

def solution(nums):
    answer = 0

    for i in range(0, len(nums) - 2):
        for j in range(i + 1, len(nums) - 1):
            for k in range(j + 1, len(nums)):
                if isPrime(nums[i], nums[j], nums[k]): answer += 1

    return answer

# from itertools import combinations as cb

# def solution(nums):
#     answer = 0
#     for a in cb(nums, 3):
#         cand = sum(a)
#         for j in range(2, cand):
#             if cand % j == 0:
#                 break
#         else:
#             answer += 1
#     return answer

File: Programmers/Lv.1.Kotlin/소수만들기.kt
import kotlin.math.*

class Solution {
    fun solution(nums: IntArray): Int {
        var answer = 0
 
        for (i in 0 until nums.size - 2) {
            for (j in (i + 1) until nums.size - 1) {
                for (k in (j + 1) until nums.size) {
                    if (isPrime(nums[i] + nums[j] + nums[k])) {
                        answer++
                    }
                }
            }
        }

        return answer
    }
    
    fun isPrime(n: Int): Boolean {
        if (n == 1) return false
        
        for (i in 2..sqrt(n.toDouble()).toInt()) {
            if (n % i == 0) return false
        }
        
        return true
    }
}